src/backend/columnar/columnar_customscan.c

```diff
@@ -816,7 +1035,15 @@ ColumnarScan_EndCustomScan(CustomScanState *node)
 static void
 ColumnarScan_ReScanCustomScan(CustomScanState *node)
 {
+	CustomScan *cscan = (CustomScan *) node->ss.ps.plan;
+	ColumnarScanState *columnarScanState = (ColumnarScanState *) node;
+
+	ResetExprContext(columnarScanState->css_RuntimeContext);
+	EvaluateClauseParams(columnarScanState->css_RuntimeContext,
+						 cscan->custom_exprs);
```

Comment:
The result of the mutator is not stored anywhere. Shouldn't this be stored in `columnarScanState->qual`?

From my understanding the mutator will _always_ return a copy, although the name suggest it mutates and object it doesn't really change the original object.

It feels that this currently doesn't perform any meaningful work.

Reply:
That was a mistake in this version. Thank you, fixed. I added some tests (only a couple for now, but I'll add more) and that should prevent this kind of thing from appearing again.